eclass/zig.eclass

```diff
@@ -0,0 +1,565 @@
+#
+# System mode is toggled here and not in "src_unpack" because they
+# could have been fetched by "live_fetch" in live ebuilds instead.
+zig_src_prepare() {
+	default_src_prepare
+
+	mkdir -p "${BUILD_DIR}" || die
+	einfo "BUILD_DIR: \"${BUILD_DIR}\""
+
+	local system_dir="${ZBS_ECLASS_DIR}/p/"
+	mkdir -p "${system_dir}" || die
+	ZBS_ARGS_BASE+=( --system "${system_dir}" )
+}
+
+# @FUNCTION: zig_src_configure
+# @DESCRIPTION:
+# Creates ZBS_ARGS array which can be used in all future phases,
+# by combining ZBS_ARGS_BASE set previously, my_zbs_args from ebuild,
+# and ZBS_ARGS_EXTRA by user, in this order.
+#
+# Specific flags currently only add support for the cross-compilation.
+# They are likely to be extended in the future.
+zig_src_configure() {
+	# Handle quoted whitespace.
+	eval "local -a ZBS_ARGS_EXTRA=( ${ZBS_ARGS_EXTRA} )"
+
+	# Since most arguments in array are also cached by ZBS, we
+	# want to reuse array as much as possible, so prevent
+	# modification of it.
+	declare -g -a -r ZBS_ARGS=(
+		# Base arguments from pkg_setup/setup_base_args
+		"${ZBS_ARGS_BASE[@]}"
+
+		# Arguments from ebuild
+		"${my_zbs_args[@]}"
+
+		# Arguments from user
+		"${ZBS_ARGS_EXTRA[@]}"
+	)
+
+	einfo "ZBS: configured with:"
+	einfo "${ZBS_ARGS[@]}"
+}
+
+# @FUNCTION: zig_src_compile
+# @USAGE: [<args>...]
+# @DESCRIPTION:
+# Calls "ezig build" with previously set ZBS_ARGS.
+# Args passed to this function will be passed after ZBS_ARGS.
+zig_src_compile() {
+	pushd "${BUILD_DIR}" > /dev/null || die
+
+	local args=( "${ZBS_ARGS[@]}" "${@}" )
+	einfo "ZBS: compiling with: ${args[@]}"
+	nonfatal ezig build "${args[@]}" || die "ZBS: compilation failed"
+
+	popd > /dev/null || die
+}
+
+# @FUNCTION: zig_src_test
+# @USAGE: [<args>...]
+# @DESCRIPTION:
+# If "test" step exist, calls "ezig build test" with previously set
+# ZBS_ARGS.
+# Args passed to this function will be passed after ZBS_ARGS.
+# Note: currently step detection might give false positives in
+# very rare cases, it will be improved in the future.
+zig_src_test() {
+	pushd "${BUILD_DIR}" > /dev/null || die
+
+	local args=( "${ZBS_ARGS[@]}" "${@}" )
+
+	# UPSTREAM std.testing.tmpDir and a lot of other functions
+	# do not respect --cache-dir or ZIG_LOCAL_CACHE_DIR:
+	# https://github.com/ziglang/zig/issues/19874
+	mkdir ".zig-cache/" || die
+
+	local found_test_step=false
+
+	local -a steps
+	readarray steps < <(nonfatal ezig build --list-steps "${args[@]}" ||
+		die "ZBS: listing steps failed")
+
+	local step
+	for step in "${steps[@]}"; do
+		# UPSTREAM Currently, step name can have any characters in it,
+		# including whitespaces, so splitting names and descriptions
+		# by whitespaces is not enough for some cases.
+		# We probably need something like  "--list-steps names_only".
+		# In practice, almost nobody sets such names.
+		local step_name="$(awk '{print $1}' <<< "${step}")"
+		if [[ "${step_name}" == test ]]; then
+			found_test_step=true
+			break
+		fi
+	done
```

Comment:
Sorry for missing this earlier, but isn't this basically a glorified:

```
grep '^test\s' < <(nonfatal ...)
```

(or alike, don't know what the exact output is)?

Reply:
fda5525fd68b01b8548a8e1458b5bddfa9f6a7b4 . Output is like this:

```
  install (default)            Copy build artifacts to prefix path
  uninstall                    Remove build artifacts from prefix path
  run                          Run the app
  test                         Run unit tests
```